### ISSUE TITLE
Not getting DB traces for java Vertx Version: 3.9.2

### ISSUE THREAD

**Issue by dwipeshmishra** [open-telemetry/opentelemetry-java-instrumentation#7018]
**Describe the bug**
Not getting DB traces for java `Vertx Version: 3.9.2` while auto-instrumenting application with opentelemetry but getting DB traces using same configuration for `Vertx Version: 3.7.1`

**Steps to reproduce**
Otel Configuration:
-Dotel.resource.attributes=service.name=demo-service -Dotel.exporter.otlp.endpoint=http://localhost:4317 -Dotel.javaagent.debug=false -Dotel.metrics.exporter=none -Dotel.traces.exporter=otlp -Dotel.integration.jdbc-datasource.enabled=true

**What did you expect to see?**
Was expecting to get mysql db traces but not getting in debug logs.

**What did you see instead?**
DB traces were missing
[vertxAppWithoutDBTraces.txt](https://github.com/open-telemetry/opentelemetry-java-instrumentation/files/9907352/vertxAppWithoutDBTraces.txt)

**What version are you using?**
Otel Version: 1.18.0

**Environment**
Compiler: 11.0.15
OS: Mac OS 12.5.1

**Comment by trask** [comment]
hi @dwipeshmishra! can you attach an example from the debug logs of the jdbc telemetry that you are seeing under vertx 3.7.1?

**Comment by dwipeshmishra** [comment]
@trask Debug logs for Vertx Version: 3.7.1
[vertxAppWithDBTraces.txt](https://github.com/open-telemetry/opentelemetry-java-instrumentation/files/9917499/vertxAppWithDBTraces.txt)

**Comment by jsuereth** [comment]
@trask Do we implement instrumentation for https://github.com/eclipse-vertx/vertx-sql-client yet?   I think this might be a feature request.

**Comment by trask** [comment]
ah, no we don't 👍

**Comment by PhilKes** [comment]
@dwipeshmishra from what I see they already wrote an otel-instrumentation for their http and sql clients [vertx-tracing](https://github.com/eclipse-vertx/vertx-tracing) outside of this repo.

I tried it out locally with `vertx-mysql-client:4.3.8` using https://vertx.io/docs/vertx-opentelemetry/java/ and the span gets created:
![image](https://user-images.githubusercontent.com/39240633/221209511-6541e61f-d552-432d-94e2-5ccc1042d75d.png)

**Comment by breedx-splk** [comment]
Yes, this instrumentation exists now. Please reopen if something is wrong/incomplete.
